src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,319 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->fallbackPool->clear();
+
+        return empty($this->values);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        if ($this->hasItemInternally($key)) {
+            return false;
+        }
+
+        return $this->fallbackPool->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $result = true;
+
+        foreach ($keys as $key) {
+            $deleted = $this->deleteItem($key);
+            $result = $result ? $deleted : false;
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        if ($this->hasItemInternally($item->getKey())) {
+            return false;
+        }
+
+        return $this->fallbackPool->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        if ($this->hasItemInternally($item->getKey())) {
+            return false;
+        }
+
+        return $this->fallbackPool->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        return $this->fallbackPool->commit();
+    }
+
+    /**
+     * Load the cache file.
+     */
+    private function initialize()
+    {
+        if (!@include $this->file) {
+            $this->values = array();
+        }
+    }
+
+    /**
+     * Validate a key.
+     *
+     * @param string $key
+     *
+     * @return string
+     */
+    private function validateKey($key)
+    {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf('Cache key must be string, "%s" given', is_object($key) ? get_class($key) : gettype($key)));
+        }
+        if (!isset($key[0])) {
+            throw new InvalidArgumentException('Cache key length must be greater than zero');
+        }
+        if (isset($key[strcspn($key, '{}()/\@:')])) {
+            throw new InvalidArgumentException('Cache key contains reserved characters {}()/\@:');
+        }
+
+        return $key;
+    }
+
+    /**
+     * Generator for items.
+     *
+     * @param array $keys
+     *
+     * @return \Generator
+     */
+    private function generateItems(array $keys)
+    {
+        $f = $this->createCacheItem;
+        $fallbackKeys = array();
+
+        foreach ($keys as $key) {
+            if ($this->hasItemInternally($key)) {
+                if ($this->storeSerialized) {
+                    $value = unserialize($this->values[$key]);
+                } else {
+                    $value = $this->values[$key];
+                }
+
+                yield $key => $f($key, $value, true);
+            } else {
+                $fallbackKeys[] = $key;
+            }
+        }
+
+        foreach ($this->fallbackPool->getItems($fallbackKeys) as $item) {
```

Comment:
we should wrap this foreach in an `if` statement to prevent calling getItems for nothing